src/parser/Parser.ts

```diff
@@ -384,9 +384,8 @@ export class Parser {
                 if (this.check(TokenKind.Function, TokenKind.Sub) || (this.check(TokenKind.Identifier, ...AllowedProperties) && this.checkNext(TokenKind.LeftParen))) {
                     let funcDeclaration = this.functionDeclaration(false);
 
-                    //remove this function from the lists because it's a class method
-                    this._references.functionExpressions.pop();
```

Comment:
Nice, I see you're no longer popping the functionExpressions list. This is because currently we still want to have a direct way to walk all class methods without having to walk through the class AST nodes directly.